src/Symfony/Component/DomCrawler/Tests/CrawlerTest.php

```diff
@@ -56,13 +60,23 @@ public function testAdd()
     /**
      * @expectedException \InvalidArgumentException
      */
-    public function testAddInvalidNode()
+    public function testAddInvalidType()
     {
         $crawler = new Crawler();
         $crawler->add(1);
     }
 
     /**
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage Nodes set in a Crawler must be DOMElement or DOMDocument instances, "DOMNode" given.
+     */
+    public function testAddInvalidNode()
+    {
+        $crawler = new Crawler();
+        $crawler->add(new \DOMNode());
+    }
```

Comment:
this is why I said I would rebase to fix conflicts: the order of triggers should be reverted (first the validation, then the deprecation). I will send a PR